packages/x-charts/src/ChartsTooltip/ChartsTooltip.tsx

```diff
@@ -133,14 +129,18 @@ function ChartsTooltip<T extends ChartSeriesType>(inProps: ChartsTooltipProps<T>
   });
   const { trigger = 'axis', itemContent, axisContent, slots, slotProps } = props;
 
-  const mousePosition = useMouseTracker();
+  const svgRef = useSvgRef();
+  const pointerType = usePointerType();
+
+  const popperRef: PopperProps['popperRef'] = React.useRef(null);
+  const virtualElement = React.useRef(generateVirtualElement({ x: 0, y: 0 }));
```

Comment:
For these type of cases the `useLazyRef` hook from the `@mui/utils` is useful. It allows to separate initialization and rendering, so the `generateVirtualElement({ ... })` doesn't need to run at every render (creates GC pressure for no reason).

Something like this:

```js
function createVirtualElement() {
  return generateVirtualElement({ x: 0, y: 0 })
}

// then here
const virtualElement = useLazyRef(createVirtualElement)
```

Reply:
I added it. But the react compiler seems to do not hunderstand that `useLazyRef` returns a ref. So I had to do the lazy implemetation directly

Reply:
The react compiler or eslint? From what I read on the compiler I would have expected it to be smart enough for that.

Reply:
The eslint(react-compiler/react-compiler)

When I do `virtualElement.current = ....`
> Mutating a value returned from a function whose return value should not be mutated eslint(react-compiler/react-compiler)

And when I don't put `virtualElement` in the dependency array of the `useEffect`

> React Hook React.useEffect has a missing dependency: 'virtualElement'. Either include it or remove the dependency array.eslint[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)

For the first one, I could disable eslint on the line.
For the second one, I just realized I could add the `virtualElement` to let dependency array since it is a ref it will never trigger the effect